api/types/swarm/task.go

```diff
@@ -91,13 +91,20 @@ type TaskSpec struct {
 	Runtime RuntimeType `json:",omitempty"`
 }
 
-// Resources represents resources (CPU/Memory).
+// Resources represents resources (CPU/Memory) which can be advertised by a
+// node and requested to be reserved for a task.
 type Resources struct {
 	NanoCPUs         int64             `json:",omitempty"`
 	MemoryBytes      int64             `json:",omitempty"`
 	GenericResources []GenericResource `json:",omitempty"`
 }
 
+// Limit describes limits on resources which can be requested by a task.
+type Limit struct {
```

Comment:
@tonistiigi @cpuguy83 I updated this from `ResourceLimit` to `Limit` as we discussed, but I kept `Resources` (we discussed renaming to `Reservation`, but the same type is used both for _nodes_ as well as for "reservation" for Services.